src/System.CodeDom/src/System/CodeDom/Compiler/TempFiles.cs

```diff
@@ -0,0 +1,171 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections;
+using System.IO;
+using System.Security.Permissions;
+
+namespace System.CodeDom.Compiler
+{
+    public class TempFileCollection : ICollection, IDisposable
+    {
+        private string _basePath;
+        private readonly string _tempDir;
+        private readonly Hashtable _files;
```

Comment:
I left this one as-is for now.  It's probably safe to change, but I'd want to verify that all of the various interface implementations and argument checking all behaves as expected, and there's no good test coverage for that right now.  Feel free to submit a PR that adds tests and then changes the type accordingly.